src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -745,6 +753,50 @@ private async Task SetDefaultProjectNameAsync()
             }
         }
 
+        private async Task<IVsProjectJsonToPackageReferenceMigrateResult> ExecuteUpgradeProjectJsonNuGetProjectCommandAsync(NuGetProject nuGetProject)
+        {
+            IVsProjectJsonToPackageReferenceMigrateResult migrationResult = null;
+            if (nuGetProject is not ProjectJsonNuGetProject)
+            {
+                //MessageHelper.ShowWarningMessage(Resources.ProjectJsonMigrateErrorMessage, Resources.ErrorDialogBoxTitle);
+                return null;
+            }
+
+            // TODO ?
+            //// Close NuGet Package Manager if it is open for this project
+            //IVsWindowFrame windowFrame = await FindExistingWindowFrameAsync(project);
+            //windowFrame?.CloseFrame((uint)__FRAMECLOSE.FRAMECLOSE_SaveIfDirty);
+            string projectJsonUniqueName = string.Empty;
+            if (nuGetProject.TryGetMetadata(NuGetProjectMetadataKeys.UniqueName, out string value))
+            {
+                projectJsonUniqueName = value;
+            }
+
+            string projectFullPath = string.Empty;
+            if (nuGetProject.TryGetMetadata(NuGetProjectMetadataKeys.FullPath, out string valuePath))
+            {
+                projectFullPath = valuePath;
+            }
+
+            var result = await _projectJsonMigrator.Value.MigrateProjectJsonToPackageReferenceAsync(projectFullPath);
+
+            if (result is IVsProjectJsonToPackageReferenceMigrateResult migratorResult)
+            {
+                migrationResult = migratorResult;
+
+                if (!migrationResult.IsSuccess)
+                {
+                    MessageHelper.ShowWarningMessage(migrationResult.ErrorMessage, "Migration Failed");
+                }
+                else
+                {
+                    MessageHelper.ShowInfoMessage("Migration Succeeded", "Migration Succeeded");
```

Comment:
this is modal right?
Imagine you have 2 projects being migrated, would it even work?